pkg/services/govmomi/util.go

```diff
@@ -271,39 +272,39 @@ func reconcileVSphereVMOnTaskCompletion(ctx *context.VMContext) {
 	})
 }
 
-func reconcileVSphereVMOnFuncCompletion(ctx *context.VMContext, waitFn func() (loggerKeysAndValues []interface{}, _ error)) {
-	obj := ctx.VSphereVM.DeepCopy()
+func reconcileVSphereVMOnFuncCompletion(_ context.Context, vmCtx *capvcontext.VMContext, waitFn func() (loggerKeysAndValues []interface{}, _ error)) {
+	obj := vmCtx.VSphereVM.DeepCopy()
 	gvk := obj.GetObjectKind().GroupVersionKind()
 
 	// Wait on the function to complete in a background goroutine.
 	go func() {
 		loggerKeysAndValues, err := waitFn()
 		if err != nil {
-			ctx.Logger.Error(err, "failed to wait on func")
+			vmCtx.Logger.Error(err, "failed to wait on func")
 			return
 		}
 
 		// Once the task has completed (successfully or otherwise), trigger
 		// a reconcile event for the associated resource by sending a
 		// GenericEvent into the event channel for the resource type.
-		ctx.Logger.Info("triggering GenericEvent", loggerKeysAndValues...)
-		eventChannel := ctx.GetGenericEventChannelFor(gvk)
+		vmCtx.Logger.Info("triggering GenericEvent", loggerKeysAndValues...)
+		eventChannel := vmCtx.GetGenericEventChannelFor(gvk)
 		eventChannel <- event.GenericEvent{
 			Object: obj,
 		}
 	}()
 }
 
-func reconcileVSphereVMOnChannel(ctx *context.VMContext, waitFn func() (<-chan []interface{}, <-chan error, error)) {
-	obj := ctx.VSphereVM.DeepCopy()
+func reconcileVSphereVMOnChannel(vmCtx *capvcontext.VMContext, waitFn func() (<-chan []interface{}, <-chan error, error)) {
```

Comment:
We will also need a context.Context here for the logging PR


```suggestion
func reconcileVSphereVMOnChannel(_ context.Context, vmCtx *capvcontext.VMContext, waitFn func() (<-chan []interface{}, <-chan error, error)) {
```

Reply:
@chrischdi , I add the `context.Context` for `reconcileVSphereVMOnChannel` func, pls help review it again, thanks!